app/models/miq_report/generator.rb

```diff
@@ -164,8 +164,8 @@ def queue_generate_table(options = {})
     if sync
       _async_generate_table(task.id, options)
     else
-      MiqQueue.put(
-        :queue_name  => "reporting",
```

Comment:
This was most likely the cause of the reporting bug where reporting work would be enqueued for a zone without a reporting role.  The absence of the `:role => "reporting"` and `:zone => nil` causes this message to be queued for `:zone => Zone.my_zone` (the current server's zone)